[libplist/libusbmuxd/libimobiledevice/libideviceactivation/usbmuxd] Update version

JackBoosY opened the pull request:
- Fix pkgconfig files
- Rename part of library names
- Fix definitions
- Fix find dependencies

Continue PR #13558.

Commit:
[libplist] Update to 1.3.6 and fix pkgconfig

Commit:
[libplist] Update to 1.2.219 and fix pkgconfig

Commit:
[libimobiledevice] Update to 1.3.6 and fix pkgconfig

Commit:
[usbmuxd] Update to 1.2.235 and fix definitions

Commit:
[libimobiledevice] Fix build

Commit:
[libgpod] Fix find libplist

Commit:
[libideviceactivation] Update to 1.2.235 and fix find dependencies

JackBoosY commented on ports/libideviceactivation/CONTROL:
```diff
@@ -1,4 +1,5 @@
 Source: libideviceactivation
-Version: 1.2.137
+Version: 1.2.235
+Port-Version: 1
```
```suggestion
```

JackBoosY committed:
Update ports/libideviceactivation/CONTROL

BillyONeal commented on ports/libimobiledevice/fix-functions-declaration.patch:
```diff
@@ -0,0 +1,102 @@
+ 	return COMPANION_PROXY_E_UNKNOWN_ERROR;
+ }
+ 
+-LIBIMOBILEDEVICE_API companion_proxy_error_t companion_proxy_client_new(idevice_t device, lockdownd_service_descriptor_t service, companion_proxy_client_t * client)
```
Why are you stripping LIBIMOBILEDEVICE_API ? There's likely calling convention stuff etc. going on there...

JackBoosY replied:
@BillyONeal Because this macro has been added in the function declaration.

BillyONeal commented on ports/libplist/CMakeLists.txt:
```diff
@@ -1,33 +1,79 @@
 
-if(WIN32)
+if (WIN32)
+    add_library(plist++ STATIC ${PLISTPP_SRC} ${PLIST_HEADER} ${LIBCNARY_SRC} ${LIBCNARY_HEADER})
+else()
+    add_library(plist++ ${PLISTPP_SRC} ${PLIST_HEADER} ${LIBCNARY_SRC} ${LIBCNARY_HEADER})
```
Why does only win32 get `STATIC`?

JackBoosY replied:
@BillyONeal This library doesn't export any functions, so I force change it as a static library.

BillyONeal replied:
Should it always be static then, not just under win32?

JackBoosY replied:
It should works fine in *inx.

JackBoosY commented:
Should be okay.

BillyONeal commented:
> Should be okay.

Sorry, I added comments but forgot to publish them.

JackBoosY commented on ports/libplist/CMakeLists.txt:
```diff
@@ -1,33 +1,79 @@
+set_target_properties(plist PROPERTIES OUTPUT_NAME plist-2.0)
 
-if(WIN32)
+if (WIN32)
+    add_library(plist++ STATIC ${PLISTPP_SRC} ${PLIST_HEADER} ${LIBCNARY_SRC} ${LIBCNARY_HEADER})
+else()
+    add_library(plist++ ${PLISTPP_SRC} ${PLIST_HEADER} ${LIBCNARY_SRC} ${LIBCNARY_HEADER})
+endif()
```
```suggestion
add_library(plist++ STATIC ${PLISTPP_SRC} ${PLIST_HEADER} ${LIBCNARY_SRC} ${LIBCNARY_HEADER})
```

JackBoosY committed:
Update ports/libplist/CMakeLists.txt

JackBoosY commented:
@BillyONeal Ready.

BillyONeal committed:
[libimobiledevice] Resolve incorrect linkage conflict between header and source in the other direction.

Submitted upstream as https://github.com/libimobiledevice-win32/libimobiledevice/pull/4/

BillyONeal commented:
Thanks for your contribution!
